cranelift/isle/veri/veri/src/bin/veri.rs

```diff
@@ -66,34 +70,86 @@ struct Opts {
 
 impl Opts {
     fn isle_input_files(&self) -> Result<Vec<std::path::PathBuf>> {
-        // Generate ISLE files.
-        let gen_dir = &self.work_dir;
-        generate_isle(gen_dir)?;
+        let name = self
+            .name
+            .as_ref()
+            .expect("missing ISLE compilation name");
+        
+        let codegen_dir = self
+            .codegen_crate_dir
+            .as_ref()
+            .expect("missing codegen crate directory");
+
+        let work_dir = self
+            .work_dir
+            .as_ref()
+            .expect("missing working directory");
+
+        // Generate ISLE files into work_dir
+        // let gen_dir = &self.work_dir;
+        generate_isle(work_dir)?;
 
         // Lookup ISLE compilations.
-        let compilations = get_isle_compilations(&self.codegen_crate_dir, gen_dir);
+        let compilations = get_isle_compilations(codegen_dir, work_dir);
 
         // Return inputs from the matching compilation, if any.
         Ok(compilations
-            .lookup(&self.name)
-            .ok_or(format_err!("unknown ISLE compilation: {}", self.name))?
+            .lookup(name) // now &str, not Option<String> 
```

Comment:
I would remove this comment, since moving forward there will not be a sense that it used to be different